packages/perseus/src/components/math-input.jsx

```diff
@@ -1,174 +1,50 @@
-/* eslint-disable react/sort-comp */
 // @flow
 import classNames from "classnames";
 import $ from "jquery";
 import MathQuill from "mathquill";
-import PropTypes from "prop-types";
 import * as React from "react";
 import ReactDOM from "react-dom";
 import _ from "underscore";
 
 import TexButtons from "./tex-buttons.jsx";
 
-const PT = PropTypes;
+import type {ButtonSetsType} from "./tex-buttons.jsx";
+
+type ButtonsVisibleType = "always" | "never" | "focused";
+
+type Props = {|
+    className?: string,
+    value: string,
+    onChange: $FlowFixMe,
```

Comment:
I tried to type this, but it spiraled out of control because of the weird way files like [this one](https://github.com/Khan/perseus/blob/main/packages/perseus-editor/src/widgets/interaction-editor.jsx) are handling callbacks.

https://khanacademy.atlassian.net/browse/LC-623